src/wallet/output_alias.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef __WALLET_OUTPUT_ALIAS_H__
+#define __WALLET_OUTPUT_ALIAS_H__
+
+#include "wallet/wallet.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Send transaction which creates a new alias output
+ *
+ * @param[in] w A wallet instance
+ * @param[in] sender_addr The sender address
+ * @param[in] sender_keypair The sender private key
+ * @param[in] send_amount The amount to transfer
+ * @param[in] state_ctrl_addr The receiver address
+ * @param[in] govern_addr The receiver address
+ * @param[out] alias_addr The response of the transfer
+ * @param[out] msg_res The response of the transfer
+ *
+ * @return int 0 on success
+ */
+int wallet_alias_create_send(iota_wallet_t* w, address_t* sender_addr, ed25519_keypair_t* sender_keypair,
+                             uint64_t const send_amount, address_t* state_ctrl_addr, address_t* govern_addr,
+                             address_t* alias_addr, res_send_message_t* msg_res);
+
+/**
+ * @brief Send alias state transition transaction
+ *
+ * @param[in] w A wallet instance
+ * @param[in] alias_id The alias identifier
+ * @param[in] state_ctrl_addr The state controller address
+ * @param[in] state_ctrl_keypair The state controller private key
+ * @param[in] govern_addr The governor address
+ * @param[out] msg_res The response of the transfer
+ *
+ * @return int 0 on success
+ */
+int wallet_alias_state_transition_send(iota_wallet_t* w, byte_t alias_id[], address_t* state_ctrl_addr,
+                                       ed25519_keypair_t* state_ctrl_keypair, address_t* govern_addr,
+                                       res_send_message_t* msg_res);
```

Comment:
the same as `wallet_alias_create_send`, the keypair and address are known by the wallet through the BIP44 path.
maybe rename to `wallet_alias_output_state_transition`?
```suggestion
int wallet_alias_state_transition_send(iota_wallet_t* w, byte_t alias_id[], bool state_ctrl_change, 
                                      uint32_t state_ctrl_index, address_t* govern_addr,
                                       res_send_message_t* msg_res);
```